test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -96,3 +101,14 @@
     (while (< (count (jruby-core/registered-instances pool-context))
               num-jrubies)
       (Thread/sleep 100))))
+
+(defn wait-for-jrubies-from-pool-context
+  "Wait for all jrubies to land in the pool"
+  [pool-context]
+  (let [num-jrubies (-> pool-context
+                        :pool-state
+                        deref
+                        :size)]
+    (while (< (count (jruby-core/registered-instances pool-context))
+              num-jrubies)
+      (Thread/sleep 100))))
```

Comment:
Since the instantiation of the JRuby instances should be much faster in this repo, we could probably get away with just having a `Thread/yield` here, and maybe reduce test runtime by a smidge?